File: backend/src/main/java/com/example/user/application/service/DeleteUserService.java
package com.example.user.application.service;

import com.example.user.application.port.in.DeleteUserUseCase;
import com.example.user.domain.exception.UserNotFoundException;
import com.example.user.domain.model.UserId;
import com.example.user.domain.port.UserRepository;

/**
 * Deletes a user by id.
 */
public final class DeleteUserService implements DeleteUserUseCase {

    private final UserRepository userRepository;

    public DeleteUserService(UserRepository userRepository) {
        this.userRepository = userRepository;
    }

    @Override
    public void execute(UserId id) {
        if (userRepository.findById(id).isEmpty()) {
            throw new UserNotFoundException(id);
        }
        userRepository.delete(id);
    }
}

File: frontend/src/user/application/service/DeleteUserService.ts
import { BaseMutationService } from '@/shared/application/service/BaseMutationService';
import { UserAdminGateway } from '@/user/domain/port/UserAdminGateway';
import { DeleteUserUseCase } from '@/user/application/port/in/DeleteUserUseCase';

export class DeleteUserService extends BaseMutationService<string, void> implements DeleteUserUseCase {
  constructor(private readonly gateway: UserAdminGateway) {
    super();
  }

  protected handleMutation(command: string): Promise<void> {
    return this.gateway.delete(command);
  }
}
